templates/rhdm71-optaweb-externaldb.yaml

```diff
@@ -39,67 +38,55 @@ parameters:
   name: IMAGE_STREAM_TAG
   value: "1.0"
   required: false
-- displayName: optaweb rostering persistence Dialect
-  description: optaweb rostering server persistence dialect 
-  name: OPTAWEB_ROSTERING_PERSISTENCE_DIALECT
+- displayName: optaweb rostering Persistence DS
+  description: optaweb rostering server persistence datasource (Sets the org.kie.server.persistence.ds system property)
+  name: OPTAWEB_ROSTERING_PERSISTENCE_DS
+  value: java:/jboss/datasources/optaweb
+  required: false
+## External database parameters BEGIN
+- displayName: optaweb rostering External Database Driver
+  description: optaweb rostering server external database driver
+  name: OPTAWEB_ROSTERING_EXTERNALDB_DRIVER
+  example: "mysql"
   required: true
-- displayName: optaweb rostering persistence Datasource
-  description: optaweb rostering server persistence datasource for persistence.xml 
-  name: OPTAWEB_ROSTERING_PERSISTENCE_DATASOURCE
-  example: java:/jboss/datasources/optaweb
+- displayName: optaweb rostering External Database Driver Type
+  description: optaweb rostering server external database driver type, applicable only for DB2, possible values are 4 (default) or 2
+  name: OPTAWEB_ROSTERING_EXTERNALDB_DRIVER_TYPE
+  example: "4"
+  required: false
+- displayName: optaweb rostering External Database User
+  description: optaweb rostering server external database username
+  name: OPTAWEB_ROSTERING_EXTERNALDB_USER
+  example: optaweb
   required: true
-- description: Database name
-  displayName: Database Name
-  name: RHDM_EXTERNALDB_DATABASE
+- displayName: optaweb rostering External Database Password
+  description: optaweb rostering server external database password
+  name: OPTAWEB_ROSTERING_EXTERNALDB_PWD
   required: true
-- description: Database user name
-  displayName: Database Username
-  name: RHDM_EXTERNALDB_USERNAME
+- displayName: optaweb rostering External Database URL
+  description: optaweb rostering server external database JDBC URL
+  name: OPTAWEB_ROSTERING_EXTERNALDB_URL
+  example: "jdbc:mysql://127.0.0.1:3306/optaweb"
+  required: false
+- displayName: optaweb rostering External Database Dialect
+  description: optaweb rostering server external database dialect
+  name: OPTAWEB_ROSTERING_EXTERNALDB_DIALECT
+  example: "org.hibernate.dialect.MySQL5Dialect"
   required: true
-- description: Database user password
-  displayName: Database Password
-  name: RHDM_EXTERNALDB_PASSWORD
-  value: '1'
+- displayName: optaweb rostering External Database Host
+  description: optaweb rostering server external database host
+  name: OPTAWEB_ROSTERING_EXTERNALDB_HOST
```

Comment:
HOST and PORT could not be required, if URL is set those two are not needed and vice-versa.

Reply:
@spolti This is what we do in rhpam71-kieserver-externaldb.yaml. And in fact, we will probably be removing the URL as in here: https://github.com/jboss-container-images/rhpam-7-openshift-image/pull/129/files (at least until CLOUD-2851 gets merged into cct_module which won't happen until our final sprint is done).

Reply:
In case of XA property, yes, and about the URL env for nonxa datasources?
I was thinking about it today, we could remove the XA one, but the URL for nonxa will probably work for all databases.